spec/automation/unit/method_validation/task_finished_spec.rb

```diff
@@ -27,19 +34,22 @@ def build_resolve_path
     FactoryGirl.create(:miq_provision_vmware, :provision_type => 'template',
                        :state => 'pending', :status => 'Ok',
                        :miq_request => miq_provision_request,
-                       :options => options, :userid => user.userid)
+                       :options => options, :userid => user.userid, :vm => vm)
   end
 
   it "miq_provision method succeeds" do
+    type = "automate_vm_provisioned"
+    msg = "[#{miq_server.name}] VM [] Provisioned Successfully"
     @args = "status=fred&ae_result=ok&MiqProvision::miq_provision=#{provision.id}&" \
             "MiqServer::miq_server=#{miq_server.id}&vmdb_object_type=miq_provision&" \
             "object=miq_provision&message=not used"
+    expect(notification_exists(type)).to be_nil
     add_call_method
     ws
     expect(provision.reload.status).to eq('Ok')
     expect(provision.state).to eq('finished')
-    msg = "[#{miq_server.name}] VM [] Provisioned Successfully"
     expect(miq_provision_request.reload.message).to eq(msg)
+    expect(notification_exists(type)).not_to be_nil
```

Comment:
Use `expect(Notification.find_by(:notification_type => notification_type)).not_to be_nil`

This completely remove the need for the `notification_exists` method.